src/main/java/anodot/stage/destination/AnodotTarget.java

```diff
@@ -103,6 +105,12 @@ public void destroy() {
     public void write(Batch batch) throws StageException {
         try {
             Iterator<Record> records = batch.getRecords();
+            Record lastRecord = getLastRecord(batch);
+            if (lastRecord != null && !conf.agentOffsetUrl.equals("")) {
```

Comment:
`!conf.agentOffsetUrl.equals("")` should be replaced with `!"".equals(conf.agentOffsetUrl)`
Not sure if `conf.agentOffsetUrl` can throw NullPointerException, so better to be safe